src/Bridge/Symfony/Routing/IriConverter.php

```diff
@@ -120,8 +123,20 @@ public function getIriFromItem($item, int $referenceType = UrlGeneratorInterface
 
         try {
             $identifiers = $this->generateIdentifiersUrl($this->identifiersExtractor->getIdentifiersFromItem($item), $resourceClass);
+            $currentRouteParams = $this->request->attributes->get('_route_params');
+            $pathVariables = $route = $this->router->getRouteCollection()->get($routeName)->compile()->getPathVariables();
+            if ($resourceClass === $this->request->attributes->get('_api_resource_class')) {
```

Comment:
You cannot assume that the current route is the same as the item route we got, just because the resouce class is the same.

Reply:
I know, we need check resource class & operation, i assume get operation as master item operation that put & delete have equal path and route params